manifests/repo.pp

```diff
@@ -42,6 +48,21 @@
 
   $enabled_num = bool2num(str2bool("${enabled}")) # lint:ignore:only_variable_string
 
+  if $::osfamily == 'Debian' {
+    if $gpgkey_keyid != undef and $gpgkey_source != undef {
+      $gpgkey_hash = {'id' => $gpgkey_keyid, 'source' => $gpgkey_source}
+    } else {
+      $gpgkey_hash = undef
+    }
+    if $enabled_num == 1 {
```

Comment:
It is better readable if you use the initial variable `$enabled` here instead. Call the new helper variable `$enabled_debian` (or alike) and you can remove parts of the if clause.